TestRunner.py

```diff
@@ -29,42 +28,37 @@ def load_test_cases(test_json):
 
 
 class TransformationRuns(object):
-    def __init__(self, interface, name_of_transformation, load_tests=True):
-        self.transformation = None
-        self.test_cases = None
-        # iterate through the modules in the current package
+    def __init__(self, heavy=False, transformation_name=None):
+        transformation_test_cases = []
+        is_done = False
         package_dir = Path(__file__).resolve()  # --> TestRunner.py
         transformations_dir = package_dir.parent.joinpath("transformations")
         for (_, m, _) in iter_modules([transformations_dir]):
-            if m == name_of_transformation:
-                t_py = import_module(f"transformations.{m}.transformation")
-                for cls in interface.__subclasses__():
-                    if hasattr(t_py, cls.__name__):
-                        self.transformation = load(t_py, cls)
-                        if load_tests:
-                            t_js = os.path.join(transformations_dir, m, "test.json")
-                            self.test_cases = load_test_cases(t_js)
+            t_py = import_module(f"transformations.{m}")
+            t_js = os.path.join(transformations_dir, m, "test.json")
+            for name, obj in inspect.getmembers(t_py):
+                # if its not None, means test single transformation passed via command-line.
+                if transformation_name:
+                    if name == transformation_name:
+                        for test_case in load_test_cases(t_js):
+                            if test_case["class"] == transformation_name:
+                                transformation_test_cases.append(test_case)
+                        is_done = True
                         break
+                elif inspect.isclass(obj) and hasattr(obj, "heavy"):
+                    # Run all the test cases based on heavy parameter.
+                    if heavy == obj.is_heavy():
```

Comment:
If this is the case, then "all" won't run the heavy cases, right?

Reply:
It will be based on the heavy parameter, if that's true and the user has passed "all", then it will run all the transformations where the "heavy" flag is true.

Reply:
But it should run those where the heavy flag is false also.. if you want to keep the function as it is, you should call this twice when "all" is called.